### PR TITLE
Improve convenience of usage

### PR DESCRIPTION
**Pull request by rshym**
This patch improves usage of some variables:
  * removes boot_dev variable. Each bootcmd now has it's own *_dtb_load
    with specific boot device
  * added reminders to make easier to remember mapping of bootcmd and
    corresponding boot device
  * naming of variables for network boot (tftp) changed to be unified
    with other devices
  * change name of flash_loaders to flash_z_loaders so it will be not
    lost inside other flash_* names

Signed-off-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>

### CODE REVIEW

File: doc/u-boot-env.txt [hunk 1]
```diff
@@ -1,54 +1,46 @@
 Boot storage device setting
 ===================================
-In order to make it possible to boot from various storage devices,
```
Comment (andr2000) [inline doc/u-boot-env.txt]: this section becomes empty - should it be removed?

File: doc/u-boot-env.txt [hunk 2]
```diff
@@ -57,18 +49,27 @@ setenv mmc1_xenpolicy_load ext2load mmc 2:1 0x7c000000 /boot/xenpolicy
...
-
-setenv emmc_dtb_load 'ext2load mmc 1:1 0x48000000 /boot/dom0.dtb; run set_boot_dev'
+setenv bootcmd_emmc 'run emmc_xen_load; run emmc_dtb_load; run emmc_kernel_load; run emmc_xenpolicy_load; run emmc_initramfs_load; bootm 0x48080000 0x76000000 0x48000000'
+setenv emmc_dtb_load 'ext2load mmc 1:1 0x48000000 /boot/dom0.dtb; fdt addr 0x48000000; fdt resize; fdt mknode / boot_dev; fdt set /boot_dev device mmcblk0'
 setenv emmc_initramfs_load ext2load mmc 1:1 0x76000000 /boot/uInitramfs
 setenv emmc_kernel_load ext2load mmc 1:1 0x7a000000 /boot/Image
 setenv emmc_xen_load ext2load mmc 1:1 0x48080000 /boot/xen-uImage
 setenv emmc_xenpolicy_load ext2load mmc 1:1 0x7c000000 /boot/xenpolicy
 
+Reminders for bootcmd usage
```
Comment (andr2000) [inline doc/u-boot-env.txt]: s/Reminders/Leftovers

File: doc/u-boot-env.txt [hunk 3]
```diff
@@ -57,18 +49,27 @@ setenv mmc1_xenpolicy_load ext2load mmc 2:1 0x7c000000 /boot/xenpolicy
...
+===================================
+
+These variables have no practical usage, but are reminders supposed to
```
Comment (andr2000) [inline doc/u-boot-env.txt]: s/reminders/leftovers

Comment (andr2000) [reply doc/u-boot-env.txt]: s/usage/use

File: doc/u-boot-env.txt (hunk 2 above)
Comment (andr2000) [reply doc/u-boot-env.txt]: my bad

File: doc/u-boot-env.txt (hunk 3 above)
Comment (andr2000) [reply doc/u-boot-env.txt]: my bad

File: doc/u-boot-env.txt (hunk 1 above)
Comment (rshym) [reply doc/u-boot-env.txt]: Thanks, good comment. I have added information on possible settings for 'bootcmd' and add reason for using 'z_bootcmd_help_*' to this section.

File: doc/u-boot-env.txt (hunk 3 above)
Comment (rshym) [reply doc/u-boot-env.txt]: Word 'usage' replaced by 'use'. Thanks.

**Comment by andr2000** [comment]
LGTM

**Comment by rshym** [comment]
String 'setenv bootcmd run ...' added to each block: tftp, mmc0, mmc1, emmc.

**Pull request merged**